### DISCUSSION TITLE
Prevent multiple task queuing for same concurrency key

### DISCUSSION THREAD

**Discussion by gautamsi** [triggerdotdev/trigger.dev#1437] [Q&A]
I am trying to evaluate his for pg-boss replacement, have a question. 

pg-boss allows to prevent duplicate task while in queue, that means if something is already processed already it can be queued, otherwise it will just be discarded if the item is already in queue. Idempotency is not helpful as it prevents running that task forever in the same task scope

I like to be
* queue task like scrape a website
* prevent same website being queued while previous one has not finished
* able to queue this again (if needed) only if previous run has already completed.
* able to scrape other website in parallel

is this possible?

**Comment by morgs32** [comment]
I think this is called a "dropping queue": https://effect.website/docs/concurrency/queue/#dropping-queue
